### PR TITLE
Update dl1 to dl2 script (Less required memory and less required step)

### PR DESCRIPTION
**Pull request by SeiyaNozaki**
This PR updates the dl1 to dl2 step:

- Load each random forest model just before using it. Now the script loads all models at the beginning of the script and it consumes much memory. Just pass the model path to `apply_models`. This option was already implemented in `apply_models`, but not used for the script before.
- This script applies the particle classification models to the data twice for `gammaness` and `reco_type`. But `reco_type` can be set based on `gammaness` as suggested by @maxnoe here. Even, we can remove `reco_type` which will not be used for the analysis.
https://github.com/cta-observatory/cta-lstchain/pull/1077#discussion_r1104715136

### CODE REVIEW

File: lstchain/reco/dl1_to_dl2.py [hunk 1]
```diff
@@ -731,6 +730,9 @@ def apply_models(dl1,
 
     # gammaness is the prediction probability for the first class (0)
     dl2['gammaness'] = probs[:, 0]
+    col = list(classifier.classes_).index(0)
```
Comment (maxnoe) [inline lstchain/reco/dl1_to_dl2.py:733]: Nice that you get the col automtaically.. but the line above still uses the hard-coded index

**Comment by SeiyaNozaki** [comment]
Concerning the first point, it was on purpose for multiple input files...
https://github.com/cta-observatory/cta-lstchain/pull/985

So I will modify codes like

- If there is a single input file, each model will be loaded just before it is used
- If there are multiple input files, all modes will be loaded once at the beginning and there will be no reload of models for each file

**Comment by SeiyaNozaki** [comment]
I realized I didn't read carefully comments from @gabemery during the school... sorry!

**Review by gabemery — DISMISSED** [review]

File: lstchain/reco/dl1_to_dl2.py (hunk 1 above)
Comment (vuillaut) [reply lstchain/reco/dl1_to_dl2.py:733]: @SeiyaNozaki could you move line 733 up one line and use col in 732?
After that we can merge I think

**Review by vuillaut — CHANGES_REQUESTED** [review]

File: lstchain/reco/dl1_to_dl2.py (hunk 1 above)
Comment (SeiyaNozaki) [reply lstchain/reco/dl1_to_dl2.py]: sorry I forgot to implement @maxnoe's comment.. now implemented!

**Review by vuillaut — APPROVED** [review]

**Pull request merged**